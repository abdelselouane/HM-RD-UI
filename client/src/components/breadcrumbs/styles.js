import styled from 'styled-components';

export const Container = styled.div`
  grid-area: cmess;

  .breadcrumb {
    font-size: 16px;
  }
`;

export default Container;
