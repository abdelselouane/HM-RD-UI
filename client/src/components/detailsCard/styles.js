import styled from 'styled-components';

export const Card = styled.div`
  width: 85%;
  margin-bottom: 20px;

  .card-content {
    padding: 0px 25px;
  }
`;

export const Title = styled.span`
  align-items: flex-end;
  display: flex;
  justify-content: flex-start;
  width: 100%;
  border-bottom: solid #666 1px;
  padding-bottom: 15px;

  h2 {
    display: inline-block;
    color: #f96302;
    font-weight: 900;
    padding-right: 25px;
  }

  span {
    text-transform: capitalize;
  }
`;

export const Row = styled.div`
  display: flex;
  padding: 15px;
`;

export const Col = styled.div`
  ${props => props.flex && `flex: ${props.flex}`}
  display: flex;
  flex-direction: column;
`;

export const Label = styled.div`
  font-weight: 700;
  color: #666;
`;
