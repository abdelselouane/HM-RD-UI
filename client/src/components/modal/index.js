export { SuccessModal } from './orderParts/index';
export { FailureModal } from './orderParts/index';
export { ConfirmationModal } from './orderParts/index';
export { InfoModal } from './receiveOrders/index';
export { default } from './modal';
