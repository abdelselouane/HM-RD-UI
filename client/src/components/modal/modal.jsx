import React from 'react';
import PropTypes from 'prop-types';
import kebabCase from 'lodash.kebabcase';
import styled from 'styled-components';

const StyledCardModal = styled.div`
.card-content {
  ${props => (props.scrollable ? ' overflow-y: scroll; height: 450px;' : 'overflow: hidden;')}
}
`;

const Modal = ({ styles, title, content, buttons, scrollable }) => (
  <div className="modal-container">
    <StyledCardModal className="card modal" style={styles} scrollable={scrollable}>
      <div className={`card-toolbar ${title.className}`}>
        {title.icon && <i className={`icon_${title.icon}`} />}
        <span className="card-title">
          {title.text && <h2>{title.text}</h2>}
        </span>
        {title.closeModal && <i className="icon_close" onClick={title.closeModal} />}
      </div>
      <div className={`card-content ${content.className}`}>
        {content.text && <h2 className="text">{content.text}</h2>}
        {content.body}
      </div>
      {buttons.length > 0 && (
        <div className={content.body ? 'card-actions right' : 'card-footer buttons'}>
          {buttons.map((button) => (
            <button
              key={kebabCase(button.text)}
              id={kebabCase(button.text)}
              className={`button ${button.className}`}
              onClick={button.onClick}
            >
              {button.text}
            </button>
          ))}
        </div>
      )}
    </StyledCardModal>
  </div>
);

Modal.defaultProps = {
  styles: {
    width: '50em'
  }
};

Modal.propTypes = {
  styles: PropTypes.shape({
    width: PropTypes.string
  }),
  title: PropTypes.shape({
    className: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    icon: PropTypes.string,
    handleCloseModal: PropTypes.func
  }).isRequired,
  content: PropTypes.shape({
    className: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    form: PropTypes.object
  }).isRequired,
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      className: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      onClick: PropTypes.func
    })
  ).isRequired
};

export default Modal;
