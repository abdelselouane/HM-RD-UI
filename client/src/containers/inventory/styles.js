import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
background-color: ${({ theme }) => theme.white};
box-sizing: border-box;
display: flex;
flex-direction: column;
justify-content: flex-start;
height: 100%;
overflow-y: hidden;
width: 100%;


.inventory-tabs {
  height: 100%;
}

.tabs span {
  &:focus {
    outline: none !important;
  }
  
  &[role=tab]:first-child {
    border-left: none;
  }
}

[role=tabpanel] {
    display: flex;
    flex-direction: column;
    height: 95%;
    overflow: hidden;
    padding: 20px 0px;

    .text-input-container.search {
      margin: 15px 15px;
      width: 65%;

      &.dropdown {
        width: 75%;
        margin: 15px 0px;
      }
    }
}

.search-container{
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  button {
    flex: 0.15;
  }
}

.text-input-container .input-field {
  margin-top: 0em;
}

.edit-col .button {
  margin: 0px;
}

.text-input-container.search {
  flex: 0.5;

  input {
    margin-top: -6px;
  }

  .icon_search {
    bottom: 0;
    color: ${({ theme }) => theme.orange};
    font-size: 30px;
    padding: 5px 10px;
    position: absolute;
    right: 0;
    z-index: 1;
  }
}
`;

export default Container;
