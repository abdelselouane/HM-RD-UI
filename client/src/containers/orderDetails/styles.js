import styled from 'styled-components';
import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
  background-color: ${({ theme }) => theme.white};
  align-items: flex-start;
  display: flex;
  flex-direction: column;
`;

export const CardTitle = styled.div`
  align-items: flex-end;
  display: flex;
  justify-content: space-between;
  height: 50px;
  margin: 0px;
  margin-bottom: 30px;
  padding: 0px 20px;
  width: 100%;

  h2 {
    display: inline-block;
    margin-bottom: 0px;
  }
`;

export const PurchaseOrderList = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  padding-top: 20px;
  flex: 1;
  overflow-y: scroll;
  overflow-x: hidden;
  width: 100%;
`;
