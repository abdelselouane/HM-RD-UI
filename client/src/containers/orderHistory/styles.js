import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
  background-color: ${({ theme }) => theme.white};
  align-items: flex-start;
  display: flex;
  flex-direction: column;

  .title {
    padding: 0px 20px;
  }
  
  .order-history-header {
    margin: 0px;
  
    thead tr {
      margin-left: 1em;
  
      th {
        border: none;
        font-size: 18px;
  
        &:first-child {
          padding-left: 20px;
        }
      }
    }
  }
`;

export default Container;
