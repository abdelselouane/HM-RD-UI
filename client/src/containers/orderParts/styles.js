import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
background: ${({ theme }) => theme.white};
`;

export default Container;
