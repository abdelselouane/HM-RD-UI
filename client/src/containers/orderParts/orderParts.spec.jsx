/* global describe, beforeAll, beforeEach, it, expect, jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as Styles from './styles';
import { OrderParts } from './index';

import {
  CREATE_PARTS_ORDER_SUCCESS,
  CREATE_PARTS_ORDER_FAILURE
} from '../../constants/actionTypes';
import { ConfirmationModal } from '../../components/modal';

describe('<OrderParts />', () => {
  let wrapper;
  let wrapperInstance;
  let spy;
  let props;
  let part;

  beforeAll(() => {
    ReactDOM.createPortal = jest.fn();

    part = {
      partNbr: 'mockPartNumber',
      partDescription: 'mockDescription',
      brandName: 'mockbrandName',
      quanitity: 1,
      ediFlag: 'Y'
    };

    props = {
      actions: {
        searchParts: jest.fn(),
        clearResults: jest.fn(),
        addToCart: jest.fn(),
        removeFromCart: jest.fn(),
        adjustQuantity: jest.fn(),
        hideMessage: jest.fn(),
        showMessage: jest.fn(),
        showInfoMessage: jest.fn(),
        createPartsOrder: jest.fn(() => ({ type: CREATE_PARTS_ORDER_SUCCESS }))
      },
      cart: [],
      results: [],
      history: { push: jest.fn() }
    };

    spy = null;

    global.setTimeout = jest.fn();
  });

  beforeEach(() => {
    wrapper = shallow(<OrderParts {...props} />);
    wrapperInstance = wrapper.instance();
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render styles properly', () => {
    Object.values(Styles).forEach(Component => {
      expect(toJson(shallow(<Component />))).toMatchSnapshot();
    });
  });

  it('should call clearResults action when input text length is less than 1', () => {
    spy = jest.spyOn(props.actions, 'clearResults');
    wrapperInstance.handleInputChange('searchInput', 'a');
    expect(spy).toHaveBeenCalled();
    spy.mockReset();
    spy.mockRestore();
  });

  it('should call showMessage with error message searchParts action returns error', async () => {
    spy = jest.spyOn(props.actions, 'showMessage');
    wrapper.setProps({ searchError: true });
    await wrapperInstance.handleInputChange('searchInput', 'ab');
    expect(spy).toHaveBeenCalledWith(
      'System Failure: Parts search is currently unavailable'
    );
    spy.mockReset();
    spy.mockRestore();
  });

  it('should call hideMessage if searchError is false and hasErrorMessage is true', async () => {
    spy = jest.spyOn(props.actions, 'hideMessage');
    wrapper.setProps({
      searchError: false,
      notify: { title: 'System Failure: Parts search is currently unavailable' }
    });
    await wrapperInstance.handleInputChange('searchInput', 'ab');
    expect(spy).toHaveBeenCalledWith();
    spy.mockReset();
    spy.mockRestore();
  });

  it('should call searchParts action when input text length is greater than 1', () => {
    spy = jest.spyOn(props.actions, 'searchParts');
    wrapperInstance.handleInputChange('searchInput', 'ab');
    expect(spy).toHaveBeenCalled();
    spy.mockReset();
    spy.mockRestore();
  });

  it('should set workOrderNumber on id workOrderNumber when handleInputChange is called', () => {
    wrapper.setState({ workOrderNumber: '123' });
    wrapperInstance.handleInputChange('workOrderNumber', 'ab');
    expect(wrapper.state().workOrderNumber).toBe('ab');
  });

  it('should do nothing when undefined id is called with handleInputChange', () => {
    wrapper.setState({ workOrderNumber: 'value' });
    wrapperInstance.handleInputChange('mockId', 'test');
    expect(wrapper.state().workOrderNumber).toBe('value');
  });

  it('should call adjustQuantity action in handleAdjustQuantity is called', () => {
    spy = jest.spyOn(props.actions, 'adjustQuantity');
    wrapper.setProps({ cart: [part] });
    wrapperInstance.handleAdjustQuantity('mockPartNumber', 2);
    expect(spy).toHaveBeenCalledWith(0, 2);
    spy.mockReset();
    spy.mockRestore();
  });

  it('should not call adjustQuantity action in handleAdjustQuantity if cart doesn\'t contain the partNbr', () => {
    spy = jest.spyOn(props.actions, 'adjustQuantity');
    wrapper.setProps({ cart: [] });
    wrapperInstance.handleAdjustQuantity('mockPartNumber', 2);
    expect(spy).not.toHaveBeenCalledWith();
    spy.mockReset();
    spy.mockRestore();
  });

  it('should call addToCart when adding a new part to cart when handleSelectedPart called', () => {
    spy = jest.spyOn(props.actions, 'addToCart');
    wrapperInstance.handleSelectedPart(part);
    expect(spy).toHaveBeenCalledWith(part);
    spy.mockReset();
    spy.mockRestore();
  });

  it('should not call addToCart and show info message when ediFlag is N', () => {
    const partWithNonEdi = {
      partNbr: 'mockPartNumber',
      partDescription: 'mockDescription',
      brandName: 'mockbrandName',
      quanitity: 1,
      ediFlag: 'N'
    };

    const addToCartSpy = jest.spyOn(props.actions, 'addToCart');
    const showInfoMsgSpy = jest.spyOn(props.actions, 'showInfoMessage');
    const setTimeoutSpy = jest.spyOn(global, 'setTimeout');
    wrapperInstance.handleSelectedPart(partWithNonEdi);
    expect(addToCartSpy).not.toHaveBeenCalledWith(part);
    expect(showInfoMsgSpy).toHaveBeenCalledWith('Part Not Available. This part is currently unavailable to order. Please use Legacy Tool Rental to order this part.');
    expect(setTimeoutSpy).toHaveBeenCalled();
  });

  it('should call removeFromCart when removing existing part in cart when handleSelectedPart called', () => {
    spy = jest.spyOn(props.actions, 'removeFromCart');
    props.cart = [
      {
        ...part,
        partDescription: 'anotherPart',
        partNbr: '01-0123'
      },
      part
    ];

    shallow(<OrderParts {...props} />)
      .instance()
      .handleSelectedPart(part);
    expect(spy).toHaveBeenCalledWith(1);
    spy.mockReset();
    spy.mockRestore();
  });

  it('should set modal and disableSubmit on submitOrder call', () => {
    wrapper.setState({
      disableSubmit: false,
      modal: null
    });
    wrapperInstance.submitOrder();
    expect(wrapper.state().disableSubmit).toBe(true);
    expect(wrapper.state().modal.type.name).toBe('ConfirmationModal');
  });

  it('should set modal to null and disableSubmit to false on secondary click of confirmationModal', () => {
    wrapper.setState({
      workOrderNumber: '',
      disableSubmit: false,
      modal: null
    });
    wrapperInstance.submitOrder();
    expect(wrapper.state()).toMatchObject({
      disableSubmit: true,
      modal: (
        <ConfirmationModal
          onPrimaryClick={wrapperInstance.confirmationModalPrimaryClick}
          onSecondaryClick={wrapperInstance.confirmationModalSecondaryClick}
        />
      )
    });
    wrapperInstance.confirmationModalSecondaryClick();
    expect(wrapper.state()).toMatchObject({
      workOrderNumber: '',
      disableSubmit: false,
      modal: null
    });
  });

  it('should set SuccesModal on confirmationModalPrimaryClick call on successful post', async () => {
    wrapper.setProps({
      actions: {
        ...props.actions,
        createPartsOrder: jest.fn(() => ({ type: CREATE_PARTS_ORDER_SUCCESS }))
      }
    });
    wrapper.setState({
      disableSubmit: false,
      modal: null
    });

    await wrapperInstance.confirmationModalPrimaryClick();
    expect(wrapper.state().modal.type.name).toBe('SuccessModal');
  });

  it('should set the state modal to null and disableSubmit to false on SuccessModalClick', async () => {
    wrapper.setProps({
      actions: {
        ...props.actions,
        createPartsOrder: jest.fn(() => ({ type: CREATE_PARTS_ORDER_SUCCESS }))
      }
    });
    wrapper.setState({
      disableSubmit: false,
      modal: null
    });

    const historySpy = jest.spyOn(props.history, 'push');

    // PrimaryClick
    await wrapperInstance.confirmationModalPrimaryClick();
    wrapperInstance.successModalPrimaryClick();

    expect(wrapper.state()).toMatchObject({
      disableSubmit: false,
      modal: null,
      workOrderNumber: ''
    });

    expect(historySpy).toHaveBeenCalledWith('home');

    // secondary click
    await wrapperInstance.confirmationModalPrimaryClick();
    wrapperInstance.successModalSecondaryClick();

    expect(wrapper.state()).toMatchObject({
      disableSubmit: false,
      modal: null,
      workOrderNumber: ''
    });

    expect(historySpy).toHaveBeenCalledWith('orderHistory');
  });

  it('should set the state modal to null and disableSubmit to false on FailureModalClick', async () => {
    wrapper.setProps({
      actions: {
        ...props.actions,
        createPartsOrder: jest.fn(() => ({ type: CREATE_PARTS_ORDER_FAILURE }))
      }
    });
    wrapper.setState({
      disableSubmit: false,
      modal: null
    });

    const historySpy = jest.spyOn(props.history, 'push');

    // PrimaryClick
    await wrapperInstance.confirmationModalPrimaryClick();
    wrapperInstance.failureModalPrimaryClick();

    expect(wrapper.state()).toMatchObject({
      disableSubmit: false,
      modal: null,
      workOrderNumber: ''
    });

    expect(historySpy).toHaveBeenCalledWith('home');

    // secondary click
    await wrapperInstance.confirmationModalPrimaryClick();
    wrapperInstance.failureModalSecondaryClick();

    expect(wrapper.state()).toMatchObject({
      disableSubmit: false,
      modal: null,
      workOrderNumber: ''
    });

    expect(historySpy).not.toHaveBeenCalledWith('');
  });

  it('should set SuccesModal on confirmationModalPrimaryClick call on successful post', async () => {
    wrapper.setProps({
      actions: {
        ...props.actions,
        createPartsOrder: jest.fn(() => ({ type: CREATE_PARTS_ORDER_FAILURE }))
      }
    });
    wrapper.setState({
      disableSubmit: false,
      modal: null
    });

    await wrapperInstance.confirmationModalPrimaryClick();
    expect(wrapper.state().modal.type.name).toBe('FailureModal');
  });
});
