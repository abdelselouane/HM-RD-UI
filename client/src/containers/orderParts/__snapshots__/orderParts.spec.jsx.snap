// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderParts /> should render properly 1`] = `
<Fragment>
  <Connect(withContext(Page))
    animateLayout={true}
    className="order-parts"
    showHomeNav={true}
  >
    <styled.div>
      <SearchResults
        cart={Array []}
        handleInputChange={[Function]}
        handleSelectedPart={[Function]}
        isLoading={false}
        results={Array []}
      />
      <ContextProvider
        value={
          Object {
            "handleAdjustQuantity": [Function],
            "handleRemoveFromCart": [Function],
          }
        }
      >
        <PartsOrderCart
          cart={Array []}
          disableSubmit={true}
          submitOrder={[Function]}
        />
      </ContextProvider>
    </styled.div>
  </Connect(withContext(Page))>
</Fragment>
`;

exports[`<OrderParts /> should render styles properly 1`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: carea;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.12),0 2px 2px 0 rgba(0,0,0,0.24);
}

<div
  className="c0"
/>
`;

exports[`<OrderParts /> should render styles properly 2`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: carea;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.12),0 2px 2px 0 rgba(0,0,0,0.24);
}

<div
  className="c0"
/>
`;
